compiler/analyzer.ml

```diff
@@ -283,14 +304,9 @@ and check_binop env e1 op e2 =
         env', Sast.Expr(Sast.Binop(ew1', op, ew2'), result_type)
       else binop_error typ1 op typ2
 
-    (* Equality operations - overloaded, no constraining can be done *)
-    | Eq | Neq -> 
-      let is_valid_equality = function
-        | Num | Bool | String | Unconst -> true
-        | _ -> false in 
-      if is_valid_equality typ1 && is_valid_equality typ2 then 
-        env', Sast.Expr(Sast.Binop(ew1, op, ew2), Bool)
-      else binop_error typ1 op typ2
+    (* Equality operations - overloaded, no constraining can be done, can take
```

Comment:
that makes things easier because then == can't constrain, but if it can't take certain types, then there are issues say we have a func
```ocaml
do comp = (x,  y) ->
  return x == y
```
we can't constrain `x` or `y` to anything, but if we limit `==` to only nums, bools, strings, then there would still be invalid inputs to the function. We would need to come up with another type just for `==`